app/controllers/application_controller.rb

```diff
@@ -54,22 +57,41 @@ def referer_url
   end
 
   def render_liquid(liquid_layout, view)
-    return redirect_to(Settings.homepage_url) unless @page.published? || user_signed_in?
+    return redirect_to(Settings.home_page_url) unless @page.published? || user_signed_in?
     localize_by_page_language(@page)
 
     @rendered = renderer(liquid_layout).render
     @data = renderer(liquid_layout).personalization_data
     render "pages/#{view}", layout: 'member_facing'
   end
 
+  def payment_methods
+    if current_member
+      PaymentMethodFetcher.new(current_member).fetch
+    else
+      payment_method_ids = (cookies.signed[:payment_methods] || '').split(',')
+      PaymentMethodFetcher.new(recognized_member, filter: payment_method_ids).fetch
+    end
+  end
+
   def renderer(layout)
-    @renderer ||= LiquidRenderer.new(@page,       location: request.location,
-                                                  member: recognized_member,
-                                                  layout: layout,
-                                                  url_params: params)
+    @renderer ||= LiquidRenderer.new(@page, location: request.location,
+                                            member: recognized_member,
+                                            layout: layout,
+                                            url_params: params,
+                                            payment_methods: payment_methods)
```

Comment:
This is added to the personalisation object - it isn't a part of the cached markup.